x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/Verifier.java

```diff
@@ -59,20 +63,19 @@
     }
 
     private static void checkForNonCollapsableSubselects(PhysicalPlan plan, List<Failure> failures) {
-        Holder<Boolean> hasLimit = new Holder<>(Boolean.FALSE);
-        Holder<List<Order>> orderBy = new Holder<>();
-        plan.forEachUp(p -> {
-            if (hasLimit.get() == false && p instanceof LimitExec) {
-                hasLimit.set(Boolean.TRUE);
-                return;
-            }
-            if (p instanceof OrderExec) {
-                if (hasLimit.get() && orderBy.get() != null && ((OrderExec) p).order().equals(orderBy.get()) == false) {
```

Comment:
Related to the removed query comment: can't we keep this "pass-through" check? I.e. chaining multiple orderings if they aren't combined with other clauses?